docs: `/api/plan` and `/api/apply` endpoints (#3013)
runatlantis/atlantis: runatlantis.io/docs/server-configuration.md

  Only enable in trusted settings.
  :::

### `--api-secret`
  ```bash
  atlantis server --api-secret="secret"
  # or (recommended)
  ATLANTIS_API_SECRET="secret"
  ```

>   Secret used to validate requests made to the [API endpoints](api-endpoints.html). If not present, the api endpoints will not be accessible.

Reviewer: ```suggestion
  Required secret used to validate requests made to the [`/api/*` endpoints](api-endpoints.html).
```
